### PR TITLE
DR-790 Clear filter data

### PR DESCRIPTION
**Pull request by myessail**
clearbutton
clearcontext

### CODE REVIEW

**Review by mackenziekerwin — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -240,7 +241,7 @@
   },
   "husky": {
     "hooks": {
-      "post-merge": "node tools update"
+      "post-merge": "npm install"
```
Comment (fboulnois) [inline package.json:244]: Why did you need to switch?

File: src/components/dataset/query/sidebar/filter/CategoryFilter.jsx [hunk 2]
```diff
@@ -24,14 +24,14 @@ export class CategoryFilter extends React.PureComponent {
...
-    count: PropTypes.string,
+    count: PropTypes.number,
     filterData: PropTypes.object,
     handleChange: PropTypes.func,
     name: PropTypes.string,
     table: PropTypes.string,
   };
 
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (fboulnois) [inline src/components/dataset/query/sidebar/filter/CategoryFilter.jsx]: Is the UNSAFE tag something that was added automatically? What caused this change?

**Review by fboulnois — CHANGES_REQUESTED** [review]

File: src/components/dataset/query/sidebar/filter/CategoryFilter.jsx (hunk 2 above)
Comment (thathert) [reply src/components/dataset/query/sidebar/filter/CategoryFilter.jsx]: It's recommended we switch to [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) instead of this. Let's try it out with this PR.

File: src/components/dataset/query/sidebar/filter/CategoryFilterGroup.jsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ export class CategoryFilterGroup extends React.PureComponent {
   render() {
     const { column, filterData, values, table, originalValues } = this.props;
     const checkboxes = _.keys(originalValues).map(name => {
-      const count = values.hasOwnProperty(name) ? values[name] : 0;
+      const count = values.hasOwnProperty(name) ? parseInt(values[name], 10) : 0;
```
Comment (thathert) [inline src/components/dataset/query/sidebar/filter/CategoryFilterGroup.jsx:54]: 👍

**Review by thathert — CHANGES_REQUESTED** [review]
I like the approach, but the `UNSAFE_` makes me sad.

File: package.json (hunk 1 above)
Comment (myessail) [reply package.json:244]: this is updating packages every time i switch branches, which is becoming more of an annoyance than something helpful. we're going to start auto-merging dependabot PRs once we have end to end tests (next sprint) so I'm not too concerned about it

**Review by fboulnois — APPROVED** [review]
lgtm

**Review by thathert — COMMENTED** [review]
Thanks for fixing!
